File: python/101-150/111. Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
import sys
class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        queue = [root]
        depth = 0
        while queue:
            temp = []
            for node in queue:
                if not node.left and not node.right:
                    return depth + 1
                if node.left:
                    temp.append(node.left)
                if node.right:
                    temp.append(node.right)
            depth += 1
            queue = temp
        

File: cpp/101-150/111. Minimum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
 #include<climits>
class Solution {
public:
	int minDepth(TreeNode* root) {
		if(!root)
			return 0;
		if(!root->left && !root->right)
			return 1;
		int left, right;
		left = right = INT_MAX;
		if(root->left)
			left = minDepth(root->left);
		if(root->right)
			right = minDepth(root->right);
		return min(left, right) + 1;
	}
};